src/filter/mod.rs

```diff
@@ -538,19 +538,56 @@ fn unapply2<'a>(
             }
             let blob = &format!("{}{}\n", &blob, pretty(parsed, 0));
 
-            let tree = if mapped != "" {
+            // Remove workspace.josh from the tree to prevent it from being parsed again
+            // further down the callstack leading to endless recursion.
+            let tree = tree::insert(
+                &transaction.repo(),
+                &tree,
+                &Path::new("workspace.josh"),
+                git2::Oid::zero(),
+                0o0100644,
+            )?;
+
+            // Insert a dummy file to prevent the directory from dissappearing through becoming
+            // empty.
+            let tree = tree::insert(
+                &transaction.repo(),
+                &tree,
+                &Path::new("DUMMY-df97a89d-b11f-4e1c-8400-345f895f0d40"),
```

Comment:
where does this number come from?

Reply:
https://www.uuidgenerator.net/version4

Reply:
It does not mean anything. just a filename that it unlikely to present in the tree by accident.

Reply:
ok